pkg/tsdb/models.go

```diff
@@ -85,3 +88,88 @@ func NewTimeSeries(name string, points TimeSeriesPoints) *TimeSeries {
 		Points: points,
 	}
 }
+
+type DataFrames interface {
+	Encoded() ([][]byte, error)
+	Decoded() (data.Frames, error)
+}
+
+type decodedDataFrames struct {
+	decoded data.Frames
+	encoded [][]byte
+}
+
+func NewDecodedDataFrames(decodedFrames data.Frames) DataFrames {
```

Comment:
Perhaps NewDataFrames, with a type switch might be more straight forward when consuming the package?

Reply:
How do you mean, taking an interface as argument?

Reply:
Yes, where the interface{} will be valid in the type switch at runtime if it is [][]byte or data.Frames

Reply:
I rather have typed contracts that shows the intent from usage.